### PR TITLE
[SR 5641] Implement support for defer in the playground transform and PC macro.

### PR DESCRIPTION
**Pull request by cwakamo**
Implements handling of `defer` statements in both the playground transform and PC macro.

Original patch for the playground transform provided by @swiftix.

Resolves [SR-5641](https://bugs.swift.org/browse/SR-5641) and &lt;rdar://problem/33764082&gt;.

### CODE REVIEW

**Comment by cwakamo** [comment]
@swift-ci Please test

**Comment by cwakamo** [comment]
@jrose-apple Feel free to add alternate reviewers if someone else should look at this code too.

File: lib/Sema/PCMacro.cpp [hunk 1]
```diff
@@ -282,6 +284,18 @@ class Instrumenter : InstrumenterBase {
     }
     return DCS;
   }
+  
+  DeferStmt *transformDeferStmt(DeferStmt *DS) {
+    if (auto *FD = DS->getTempDecl()) {
+      auto Implicit = FD->isImplicit();
+      FD->setImplicit(false);
```
Comment (jrose-apple) [inline lib/Sema/PCMacro.cpp]: I don't love this hack, but I guess it's okay. Mind commenting *why* we have to do it, though?

File: lib/Sema/PlaygroundTransform.cpp [hunk 2]
```diff
@@ -153,6 +155,17 @@ class Instrumenter : InstrumenterBase {
     }
   }
 
+  DeferStmt *transformDeferStmt(DeferStmt *DS) {
```
Comment (jrose-apple) [inline lib/Sema/PlaygroundTransform.cpp]: …definitely a candidate for future refactoring up into InstrumenterBase.

**Review by jrose-apple — APPROVED** [review]

**Comment by cwakamo** [comment]
@swift-ci Please smoke test and merge

**Comment by cwakamo** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**